controllers/front/callbacknotification.php

```diff
@@ -238,4 +253,43 @@ public function postProcess()
         unlockCallback($lockFileName, $lockFileHandle);
         exit($message);
     }
+
+    public function processChildOrder($cart, $transactionId, $response, $transactionStatus, $auth_statuses, $captured_statuses, $lockFileName, $lockFileHandle)
+    {
+        $shopOrderId = $response->shopOrderId;
+        // Check if an order exist
+        $order = getChildOrderFromUniqueId($shopOrderId);
+        if (!Validate::isLoadedObject($order)) {
+            $order_id = Order::getOrderByCartId((int) ($cart->id));
```

Comment:
Casting to int can cause the result to be 0, which might be a valid order_id, causing side effects. You need to validate the integrity of the value instead. If you expect and INT, then validate the value is indeed an INT before casting.

Reply:
@emicha, we make sure the cart object and its ID are valid before using them in the code. Here's how we do it:

```
            // Load the cart
            $cart = getCartFromUniqueId($shopOrderId);
            if (!Validate::isLoadedObject($cart)) {
                exit('Could not load cart - exiting');
            }
```
This check ensures that cart->id is always valid, which prevents any potential issues with casting.